include/property.h

```diff
@@ -0,0 +1,181 @@
+#ifndef __PROPERTY_H__
+#define __PROPERTY_H__
+
+#include <iostream>
+#include <set>
+#include <tuple>
+#include <sstream>
+#include <regex>
+
+namespace clayer {
+
+    // Concepts used for property reading and writing
+    template<typename T>
+    concept bool Streamable = requires(T o, std::ostream& s) {
+        {s << o} -> std::ostream&;
+    };
+
+    template<typename T>
+    concept bool StringRef = requires(T o) {
+        {*o} -> std::string;
+    };
+
+    // template<typename T=void>
+    // void print<>(){
+    //     std::cout << "\n";
+    // }
+
+    // template<Streamable T, typename... Targs>
+    // void print(T o, Targs... tail) {
+    //     std::cout << o <<" ";
+    //     print(tail...);
+    // }
+
+    // ## Log LogRecord - Any item that can be printed in a Log Record
+    enum log_properties {
+        FILE,   // - File name (CodeContext)
+        FUNC,   // - Function name (CodeContext)
+        LINE,   // - Line number (CodeContext)
+        // LEVEL
+        // NAME
```

Comment:
What does this `NAME` field serve when we already have the hex as the unique identifier.